aries_cloudagent/wallet/base.py

```diff
@@ -55,6 +56,34 @@ async def create_key(
             WalletError: If there is another backend error
         """
 
+    @abstractmethod
+    async def assign_kid_to_key(self, verkey: str, kid: str) -> KeyInfo:
+        """Assign a KID to a key.
+
+        This is separate from the create_key method because some DIDs are only
+        known after keys are created.
+
+        Args:
+            verkey: The verification key of the keypair
+            kid: The kid to assign to the keypair
+
+        Returns:
+            A `KeyInfo` representing the keypair
+
+        """
+
+    @abstractmethod
+    async def get_key_by_kid(self, kid: str) -> KeyInfo:
+        """Fetch a key by looking up its kid.
+
+        Args:
+            kid: the key identifier
+
+        Returns:
+            The key identified by kid
+
+        """
```

Comment:
Calling out a fundamental paradigm shift starting here for ACA-Py: to work with DCv2, these changes start using full DID URL references to verification methods to identify keys instead of the base58 encoding of the public key.

And a limitation: There is only one kid associated with a key. Does it make sense for a key to be looked up by multiple different key ids?

Reply:
Now that I think on it, the way we implemented the interface, it doesn't actually preclude the ability to associated a key with multiple kids. But the implementation of the interface does impose this limitation.